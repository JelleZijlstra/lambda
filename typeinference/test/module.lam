let option_m = module
	interface {
		data option a = None | Some a,
		deoptionalize : option int -> int -> int,
		is_some : option int -> bool
	}

	data option a = None | Some a

	let deoptionalize = \o. \dflt. match o with {
		None -> dflt
		| Some v -> v
	}

	let is_some = \o. match o with {
		None -> false
		| Some _ -> true
	};;

	print 42
end in
option_m
