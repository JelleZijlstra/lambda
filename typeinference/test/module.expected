%%stdout%%
42
Result: module
	let __result__ = module
	let is_some = \o. match o with None -> false | Some _ -> true
	let deoptionalize = \o. \dflt : 'typevar1. match o with None -> dflt | Some v -> v
	let __result__ = ()
end
end
%%stderr%%
